apollo-federation/src/query_graph/graph_path/transition.rs

```diff
@@ -26,3 +55,508 @@ impl GraphPathTriggerVariant for QueryGraphEdgeTransition {
+                        let parent_type_pos_in_subgraph =
+                            subgraph_schema.try_get_type(parent_type_pos.type_name().clone());
+                        if parent_type_pos_in_subgraph.is_none()
+                            && tail_type_pos.type_name() != parent_type_pos.type_name()
+                        {
+                            // This is due to us looking for an implementation field, but the
+                            // subgraph not having that implementation because it uses
+                            // @interfaceObject on an interface of that implementation.
+                            break 'details format!(
+                                "cannot find implementation type \"{}\" (supergraph interface \"{}\" is declared with @interfaceObject in \"{}\")",
+                                parent_type_pos, tail_type_pos, subgraph,
+                            );
+                        }
+                        let Some(Ok::<CompositeTypeDefinitionPosition, _>(
+                            parent_type_pos_in_subgraph,
+                        )) = parent_type_pos_in_subgraph.map(|pos| pos.try_into())
+                        else {
+                            break 'details format!(
+                                "cannot find field \"{}\"",
+                                field_definition_position,
+                            );
+                        };
+                        let Ok(field_pos_in_subgraph) = parent_type_pos_in_subgraph
+                            .field(field_definition_position.field_name().clone())
+                        else {
+                            break 'details format!(
+                                "cannot find field \"{}\"",
+                                field_definition_position,
+                            );
+                        };
+                        let Ok(field_in_subgraph) =
+                            field_pos_in_subgraph.get(subgraph_schema.schema())
+                        else {
+                            break 'details format!(
+                                "cannot find field \"{}\"",
+                                field_definition_position,
+                            );
+                        };
+                        // The subgraph has the field but no corresponding edge. This should only
+                        // happen if the field is external.
+                        let Some(application) = field_in_subgraph
+                            .directives
+                            .get(external_directive_definition_name)
+                        else {
+                            bail!(
+                                "{} in {} is not external but there is no corresponding edge",
+                                field_pos_in_subgraph,
+                                subgraph,
+                            );
+                        };
+                        let external_reason = metadata
```

Comment:
A comment block in JS code is not ported here.
```typescript
            // but the field is external in the "subgraph-extracted-from-the-supergraph", but it might have been forced to an external
            // due to being a used-overriden field, in which case we want to amend the message to avoid confusing the user.
            // Note that the subgraph extraction marks such "forced external due to being overriden" by setting the "reason" to "[overridden]".
```
It seems worth porting. But, we need to correct some typos "overriden" => "overridden".

Reply:
Good catch, ported the comment over with corrections.